tests/clickhouse/query_dsl/test_project_id.py

```diff
@@ -176,17 +178,20 @@ def test_find_projects(
 ) -> None:
     events = get_dataset("events")
     if expected_projects is None:
-        with pytest.raises(ParsingException):
+        with pytest.raises(ValidationException):
             request = json_to_snql(query_body, "events")
             request.validate()
             query, _ = parse_snql_query(str(request.query), events)
             assert isinstance(query, Query)
+            run_entity_validators(query, HTTPQuerySettings())
             identity_translate(query)
     else:
         request = json_to_snql(query_body, "events")
         request.validate()
         query, _ = parse_snql_query(str(request.query), events)
         assert isinstance(query, Query)
+        assert isinstance(query, Query)
+        run_entity_validators(query, HTTPQuerySettings())
```

Comment:
Is this necessary?

Reply:
I just ran the test without this, I don't think you need it

Reply:
Sure we can remove it. I've added this here because I didn't want to change  what this test use to do (parse the query, run entity validators, then find the project_ids). The entity validator part contains `EntityRequiredColumnValidator` which ensures that a project_id was first passed in before running the project_id finder

Reply:
Wait, sorry we need this here. There are two tests that are expected to fail because it does not provide project_id (e.g. https://github.com/getsentry/snuba/blob/6680821603eca90018d20f8d9f64203c2086c8c9/tests/clickhouse/query_dsl/test_project_id.py#L48-L56)

Reply:
ok but that's not really testing the component that is under test. what is being tested here is the project id finder.